test/e2e/framework/job/wait.go

```diff
@@ -60,19 +59,38 @@ func WaitForJobPodsSucceeded(ctx context.Context, c clientset.Interface, ns, job
 
 // waitForJobPodsInPhase wait for all pods for the Job named JobName in namespace ns to be in a given phase.
 func waitForJobPodsInPhase(ctx context.Context, c clientset.Interface, ns, jobName string, expectedCount int32, phase v1.PodPhase, timeout time.Duration) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, timeout, false, func(ctx context.Context) (bool, error) {
-		pods, err := GetJobPods(ctx, c, ns, jobName)
-		if err != nil {
-			return false, err
-		}
-		count := int32(0)
-		for _, p := range pods.Items {
-			if p.Status.Phase == phase {
-				count++
+	return framework.Gomega().
+		Eventually(ctx, func(ctx context.Context) (*v1.PodList, error) {
```

Comment:
Use `HandleRetry` here to make sure that transient errors are retried. Also I'd consider defining the getter and matcher functions in function variables to keep consistency. You have an example [here](https://github.com/carlory/kubernetes/blob/6cb5ea56cb96f6a4ddec677d3d94faf07179368d/test/e2e/framework/job/wait.go#L89)

Reply:
will do